cloud/scope/powervs_cluster.go

```diff
@@ -2284,7 +2298,7 @@ func (s *PowerVSClusterScope) GetServiceName(resourceType infrav1beta2.ResourceT
 		if s.Network() == nil || s.Network().Name == nil {
 			return ptr.To(fmt.Sprintf("DHCPSERVER%s_Private", s.InfraCluster()))
 		}
-		return s.Network().Name
+		return ptr.To(fmt.Sprintf("DHCPSERVER%s_Private", *s.Network().Name))
```

Comment:
What made you add this, I am not able to recollect!

Reply:
In a case where `network.Name` is set in spec to a DHCP based n/w and resource exists in cloud, the flow wasn't working as expected as we check for resource with spec.name(for example: capi-test) against the name in cloud which will be in a different format(DHCPSERVERcapi-test_Private). Hence added this change to convert the name accordingly.

However, this would be wrong if user uses a different network type. Reverted this change and will need to revisit this.